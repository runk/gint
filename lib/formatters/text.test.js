const Formatter = require('./text');

describe('lib/formatters/text', () => {
  it('should work for very basic case', () => {
    const formatter = new Formatter();
    formatter.line('error', 'foo', new Error('First'));
    formatter.line('warning', 'bar', new Error('Second'));
    expect(formatter.result()).toMatchSnapshot();
  });

  it('should work for multiply errors/warnings', () => {
    const formatter = new Formatter();
    formatter.line('error', 'foo', new Error('Burito'));
    formatter.line('error', 'foz', new Error('Taco'));
    formatter.line('warning', 'bar', new Error('Nachos'));
    formatter.line('warning', 'baz', new Error('Quesadilla'));
    expect(formatter.result()).toMatchSnapshot();
  });

  it('should work when no errors', () => {
    const formatter = new Formatter();
    expect(formatter.result()).toMatchSnapshot();
  });
});
