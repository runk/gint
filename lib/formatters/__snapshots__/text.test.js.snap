// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/formatters/text should work for multiply errors/warnings 1`] = `
"Gint found some issues:
  [[31merror[39m]	[90mfoo[39m	Burito
  [[31merror[39m]	[90mfoz[39m	Taco
  [[33mwarning[39m]	[90mbar[39m	Nachos
  [[33mwarning[39m]	[90mbaz[39m	Quesadilla

[31m[1m âœ– 4 problems (2 errors, 2 warnings)[22m[39m
"
`;

exports[`lib/formatters/text should work for very basic case 1`] = `
"Gint found some issues:
  [[31merror[39m]	[90mfoo[39m	First
  [[33mwarning[39m]	[90mbar[39m	Second

[31m[1m âœ– 2 problems (1 error, 1 warning)[22m[39m
"
`;

exports[`lib/formatters/text should work when no errors 1`] = `""`;
